horace_core/sqw/coord_transform/@ortho_proj/ortho_proj.m

```diff
@@ -0,0 +1,547 @@
+    %   >> proj = ortho_proj(...,'label',labelcellstr,...)
+    %   >> proj = ortho_proj(...,'lab1',labelstr,...)
+    %                   :
+    %   >> proj = ortho_proj(...,'lab4',labelstr,...)
+    %
+    %
+    % Input:
+    % ------
+    % Projection axes are defined by two vectors in reciprocal space, together
+    % with optional arguments that control normalisation, orthogonality, labels etc.
+    % The input can be a data structure with fieldnames and contents chosen from
+    % the arguments below, or alternatively the arguments
+    %
+    % Required arguments:
+    %   u      [1x3] Vector of first axis (r.l.u.) defining cut plain and projection axes
+    %   v      [1x3] Vector of second axis (r.l.u.) defining cut plain and projection axes
+    %
+    % Optional arguments:
+    %   w           [1x3] Vector of third axis (r.l.u.) - only needed if the third
+    %               character of argument 'type' is 'p'. Will otherwise be ignored.
+    %
+    %   nonorthogonal  Indicates if non-orthogonal axes are permitted
+    %               If false (default): construct orthogonal axes u1,u2,u3 from u,v
+    %               by defining: u1 || u; u2 in plane of u and v but perpendicular
+    %               to u with positive component along v; u3 || u x v
+    %
+    %               If true: use u,v (and w, if given) as non-orthogonal projection
+    %               axes: u1 || u, u2 || v, u3 || w if given, or u3 || u x v if not.
+    %
+    %   type        [1x3] Character string defining normalisation. Each character
+    %               indicates how u1, u2, u3 are normalised, as follows:
+    %               - if 'a': projection axis unit length is one inverse Angstrom
+    %               - if 'r': then if ui=(h,k,l) in r.l.u., is normalised so
+    %                         max(abs(h,k,l))=1
+    %               - if 'p': if orthogonal projection axes:
+    %                               |u1|=|u|, (u x u2)=(u x v), (u x u3)=(u x w)
+    %                           i.e. the projections of u,v,w along u1,u2,u3 match
+    %                           the lengths of u1,u2,u3
+    %
+    %                         if non-orthogonal axes:
+    %                               u1=u;  u2=v;  u3=w
+    %               Default:
+    %                   'ppr'  if w not given
+    %                   'ppp'  if w is given
+    %
+
+    % Original author: T.G.Perring
+    %
+    %
+    properties
+        %
+    end
+    properties(Dependent)
+        u; %[1x3] Vector of first axis (r.l.u.)
+        v; %[1x3] Vector of second axis (r.l.u.)
+        w; %[1x3] Vector of third axis (r.l.u.) - used only if third character of type is 'p'
+        type;  % Character string length 3 defining normalisation. each character being 'a','r' or 'p' e.g. 'rrp'
+        nonorthogonal; % Indicates if non-orthogonal axes are permitted (if true)
+        %
+    end
+    properties(Dependent,Hidden) %TODO: all this should go with new sqw design
```

Comment:
these properties are used for compatibility with current sqw object. They should not exist in this form as part of ortho projection -- it will not expose them in new sqw design.